psi4/driver/procrouting/dft_funcs/superfuncs.py

```diff
@@ -108,6 +108,36 @@ def build_superfunctional(name, restricted):
     if core.has_option_changed("SCF", "DFT_ALPHA_C"):
         sup[0].set_c_alpha(core.get_option("SCF", "DFT_ALPHA_C"))
 
+    # customization of existing VV10 dispersion:
+    if (core.has_option_changed("SCF", "NL_DISPERSION_PARAMETERS") and sup[0].vv10_b() > 0.0):
+       nl_tuple = core.get_option("SCF", "NL_DISPERSION_PARAMETERS")
+       sup[0].set_vv10_b(nl_tuple[0])
+       if len(nl_tuple) > 1:
+          sup[0].set_vv10_c(nl_tuple[1])
+       if len(nl_tuple) > 2:
+          raise ValidationError("too many entries in NL_DISPERSION_PARAMETERS for DFT-NL")
+    # add VV10 correlation to any functional or modify existing
+    # custom procedures using name 'scf' without any quadrature grid like HF will fail and are not detected
+    if (core.has_option_changed("SCF", "DFT_VV10_B") and core.has_option_changed("SCF", "DFT_VV10_C")):
+        if(name.lower()=='hf'):
+           raise ValidationError("SCF: HF with -NL not implemented")
+        vv10_b = core.get_option("SCF", "DFT_VV10_B")
+        vv10_c = core.get_option("SCF", "DFT_VV10_C")
+        sup[0].set_vv10_b(vv10_b)
+        sup[0].set_vv10_c(vv10_c)
+    elif core.has_option_changed("SCF", "DFT_VV10_B"):
```

Comment:
Wouldn't it be more readable if these ifs were merged:
```
+    if (core.has_option_changed("SCF", "NL_DISPERSION_PARAMETERS") and core.has_option_changed("SCF", "DFT_VV10_B")):
+        raise ValidationError("SCF: Decide between NL_DISPERSION_PARAMETERS and DFT_VV10_B !!")
+    elif (core.has_option_changed("SCF", "NL_DISPERSION_PARAMETERS") and sup[0].vv10_b() > 0.0):
+        ...
+    elif (core.has_option_changed("SCF", "DFT_VV10_B"):
+        if(name.lower()=='hf'):
+           raise ValidationError("SCF: HF with -NL not implemented")
+        vv10_b = core.get_option("SCF", "DFT_VV10_B")
+        if (core.has_option_changed("SCF", "DFT_VV10_C"):
+            vv10_c = core.get_option("SCF", "DFT_VV10_C")
+        elif sup[0].vv10_c() != 0.0:
+            vv10_c = sup[0].vv10_c()
+        else:
+            core.print_out("SCF: VV10_C not specified. Using default (C=0.0093)!")
+            vv10_c = 0.0093
+        sup[0].set_vv10_b(vv10_b)
+        sup[0].set_vv10_c(vv10_c)
```

The `if (abs(sup[0].vv10_c() - 0.0) <= 1e-8)` check below only for overriding functional-specific "B" but keeping functional-specific non-default "C"? The way I read the code, unless set by `dict_func`, the `vv10_c` is initialised as 0.0:

https://github.com/psi4/psi4/blob/d5b1e133eb4fd2bebd695e8520938f14743fdd16/psi4/src/psi4/libfunctional/superfunctional.cc#L71

Why compare against a threshold?

Reply:
This is essentially testing if `c = 0`. The threshold avoids the "equal test" on two floats.

I noticed that the comments around the if block are outdated.
I can merge them, yes.